### PR TITLE
chore: Bump go to `1.20`

### PR DESCRIPTION
**Pull request by jonathan-innis**
Fixes # 

**Description**

Bump go to `1.20`

**How was this change tested?**

`make presubmit`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/scheduling/requirement.go [hunk 1]
```diff
@@ -28,10 +27,6 @@ import (
 	"github.com/aws/karpenter-core/pkg/apis/v1alpha5"
 )
 
-func init() {
```
Comment (jonathan-innis) [inline pkg/scheduling/requirement.go:31]:
```
Seed uses the provided seed value to initialize the default Source to a deterministic state. Seed values that have the same remainder when divided by 2³¹-1 generate the same pseudo-random sequence. Seed, unlike the Rand.Seed method, is safe for concurrent use.

If Seed is not called, the generator is seeded randomly at program startup.
```

https://tip.golang.org/doc/go1.20

**Review by bwagner5 — COMMENTED** [review]
can you update `test/` too?

**Review by bwagner5 — APPROVED** [review]
LGTM

**Review by bwagner5 — APPROVED** [review]
LGTM

**Pull request merged**